### ISSUE TITLE
[Bug] Press/hold value glitching out due to RingconFix applying multiple times

### ISSUE THREAD

**Issue by Jamie-Spruin** [ringrunnermg/Ringcon-Driver#24]
I'm not sure if anyone is still working on this, but I've been having an issue with the push/pull value going out of whack.

It seemed to be randomly incrementing based on the fix value.
![image](https://github.com/user-attachments/assets/42adcd2c-cd13-459b-9154-fc491bfb0e63)
If I don't have the fix value set, the value doesn't fall off like this.

From what I gather it seems to be due to some code around line 419 of Main.cpp where it retains the previous Ringcon value if it's 0. It will then take the last value and apply the fix again.

![image](https://github.com/user-attachments/assets/1f1da7f3-fc93-43ca-8733-ec5e3df2295b)

Pretty sure there just needs to be an else block after checking if the value is 0 to avoid applying it more than once.

This project seems really cool so I'd love to see it more accessible to people with wonky RingCon calibrations like mine.

**Comment by zac-cohen** [comment]
I've been working a bit on reviving this in a fork. When you say it's incrementing randomly based on the fix value, are you seeing the behavior where it kind of blows up into a huge number?

**Comment by Jamie-Spruin** [comment]
> I've been working a bit on reviving this in a fork. When you say it's incrementing randomly based on the fix value, are you seeing the behavior where it kind of blows up into a huge number?

Yes, if the over-correction happens multiple times in a row it can blow up into a huge number.
It can also just seem jittery if it only overcorrects once or twice in a row, or if the RingconFix value is low

With the RingconFix value set to 0 it didn't have those issues

It seems like an easy fix, I just couldn't find all the libraries to get it to compile since I think some of them don't exist anymore

**Comment by zac-cohen** [comment]
I see, I was able to reproduce and fix using exactly your suggestion in my latest commit https://github.com/zac-cohen/Ringcon-Driver . I included the old/missing libraries within this to build, though you do still need to handle vcpkg in classic mode for the libraries which aren't directly included (I only include sio-client and websocketpp).

The rest I was able to get this built with. I have a bunch of other features added in there, so if you only want this update I can make a separate place for you to pull the original app with just that fix.

As an added note, when you see the number blow up, it seems to be usually that the ring and the right joycon lost contact

**Comment by Jamie-Spruin** [comment]
Glad to hear it worked!

Having the fix in it's own place would be great! Hopefully the original author can fix it here as well(though I won't hold my breath since it hasn't been updated in years), but if anyone else stumbles into this issue they could probably use your fix

I originally wanted to try making a small game utilizing the ringcon, but gave up due to the push/pull value bugging out. Work and my main project are eating up all my time so I probably won't be able to get to that for a while 😂
